modules/imgproc/src/smooth.cpp

```diff
@@ -2770,47 +3185,95 @@ void vlineSmooth5N(const FT* const * src, const FT* m, int, ET* dst, int len)
 {
+    int i = 0;
     static const v_int16x8 v_128 = v_reinterpret_as_s16(v_setall_u16((uint16_t)1 << 15));
-
     v_int32x4 v_128_4 = v_setall_s32(128 << 16);
     if (len > 7)
     {
         ufixedpoint32 val[] = { (m[0] + m[1] + m[2] + m[3] + m[4]) * ufixedpoint16((uint8_t)128) };
         v_128_4 = v_setall_s32(*((int32_t*)val));
     }
-
-    int i = 0;
     v_int16x8 v_mul01 = v_reinterpret_as_s16(v_setall_u32(*((uint32_t*)m)));
     v_int16x8 v_mul23 = v_reinterpret_as_s16(v_setall_u32(*((uint32_t*)(m + 2))));
     v_int16x8 v_mul4 = v_reinterpret_as_s16(v_setall_u16(*((uint16_t*)(m + 4))));
-    for (; i < len - 7; i += 8)
+    for (; i < len - 31; i += 32)
     {
-        v_int16x8 v_src0, v_src1;
+        v_int16x8 v_src00, v_src10, v_src01, v_src11, v_src02, v_src12, v_src03, v_src13;
         v_int16x8 v_tmp0, v_tmp1;
 
-        v_src0 = v_load((int16_t*)(src[0]) + i);
-        v_src1 = v_load((int16_t*)(src[1]) + i);
-        v_zip(v_add_wrap(v_src0, v_128), v_add_wrap(v_src1, v_128), v_tmp0, v_tmp1);
+        v_src00 = v_load((int16_t*)(src[0]) + i);
+        v_src01 = v_load((int16_t*)(src[0]) + i + 8);
+        v_src02 = v_load((int16_t*)(src[0]) + i + 16);
+        v_src03 = v_load((int16_t*)(src[0]) + i + 24);
+        v_src10 = v_load((int16_t*)(src[1]) + i);
+        v_src11 = v_load((int16_t*)(src[1]) + i + 8);
+        v_src12 = v_load((int16_t*)(src[1]) + i + 16);
+        v_src13 = v_load((int16_t*)(src[1]) + i + 24);
+        v_zip(v_add_wrap(v_src00, v_128), v_add_wrap(v_src10, v_128), v_tmp0, v_tmp1);
         v_int32x4 v_res0 = v_dotprod(v_tmp0, v_mul01);
         v_int32x4 v_res1 = v_dotprod(v_tmp1, v_mul01);
-
-        v_src0 = v_load((int16_t*)(src[2]) + i);
-        v_src1 = v_load((int16_t*)(src[3]) + i);
-        v_zip(v_add_wrap(v_src0, v_128), v_add_wrap(v_src1, v_128), v_tmp0, v_tmp1);
+        v_zip(v_add_wrap(v_src01, v_128), v_add_wrap(v_src11, v_128), v_tmp0, v_tmp1);
+        v_int32x4 v_res2 = v_dotprod(v_tmp0, v_mul01);
+        v_int32x4 v_res3 = v_dotprod(v_tmp1, v_mul01);
+        v_zip(v_add_wrap(v_src02, v_128), v_add_wrap(v_src12, v_128), v_tmp0, v_tmp1);
+        v_int32x4 v_res4 = v_dotprod(v_tmp0, v_mul01);
+        v_int32x4 v_res5 = v_dotprod(v_tmp1, v_mul01);
+        v_zip(v_add_wrap(v_src03, v_128), v_add_wrap(v_src13, v_128), v_tmp0, v_tmp1);
+        v_int32x4 v_res6 = v_dotprod(v_tmp0, v_mul01);
+        v_int32x4 v_res7 = v_dotprod(v_tmp1, v_mul01);
+
+        v_src00 = v_load((int16_t*)(src[2]) + i);
+        v_src01 = v_load((int16_t*)(src[2]) + i + 8);
+        v_src02 = v_load((int16_t*)(src[2]) + i + 16);
+        v_src03 = v_load((int16_t*)(src[2]) + i + 24);
+        v_src10 = v_load((int16_t*)(src[3]) + i);
+        v_src11 = v_load((int16_t*)(src[3]) + i + 8);
+        v_src12 = v_load((int16_t*)(src[3]) + i + 16);
+        v_src13 = v_load((int16_t*)(src[3]) + i + 24);
```

Comment:
SSE/AVX instruction set has 16 vector registers only.

`v_res{0, 7}` (8) and `v_src*` (8) + external `v_mul*` + other exceeds registers pool.

Are you sure that there is no register spilling into stack?